spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -653,36 +653,30 @@ def add_electrical_series(recording: se.RecordingExtractor, nwbfile=None, metada
             if isinstance(recording.get_traces(), np.memmap):
                 n_bytes = np.dtype(recording.get_dtype()).itemsize
                 buffer_size = int(buffer_mb * 1e6) // (recording.get_num_channels() * n_bytes)
-                ephys_data = H5DataIO(
-                    DataChunkIterator(
-                        data=recording.get_traces().T,  # nwb standard is time as zero axis
-                        buffer_size=buffer_size
-                    ),
-                    compression='gzip'
+                ephys_data = DataChunkIterator(
+                    data=recording.get_traces().T,  # nwb standard is time as zero axis
+                    buffer_size=buffer_size
                 )
             else:
                 def data_generator(recording, channels_ids):
-                    #  generates data chunks for iterator
+                    # generates data chunks for iterator
                     for id in channels_ids:
                         data = recording.get_traces(channel_ids=[id]).flatten()
                         yield data
-
-                ephys_data = H5DataIO(
-                    DataChunkIterator(
-                        data=data_generator(
-                            recording=recording,
-                            channels_ids=channel_ids
-                        ),
-                        iter_axis=1,  # nwb standard is time as zero axis,
-                        maxshape=(recording.get_num_frames(), recording.get_num_channels())
+                ephys_data = DataChunkIterator(
+                    data=data_generator(
+                        recording=recording,
+                        channels_ids=channel_ids
                     ),
-                    compression='gzip'
+                    iter_axis=1,  # nwb standard is time as zero axis
+                    maxshape=(recording.get_num_frames(), recording.get_num_channels())
                 )
+
             if use_timestamps:
                 nwbfile.add_acquisition(
                     ElectricalSeries(
                         name=es_name,
-                        data=ephys_data,
+                        data=H5DataIO(ephys_data, compression="gzip"),
```

Comment:
Should we have compression by default?

Reply:
It was already included by default, I just moved the wrapper down here instead of up around the DataChunkIterators (saves a couple lines of code).

I do think we should always have it by default though; as far as I know there are only beneficial effects. If you disagree we can add an option for disabling it, though, that would be easy.